tests/mocks/MockClock.hh

```diff
@@ -0,0 +1,59 @@
+#ifndef RA_SERVICES_MOCK_CLOCK_HH
+#define RA_SERVICES_MOCK_CLOCK_HH
+#pragma once
+
+#include "services\IClock.hh"
+#include "services\ServiceLocator.hh"
+
+#include <unordered_map>
+
+namespace ra {
+namespace services {
+namespace mocks {
+
+class MockClock : public IClock
+{
+public:
+    MockClock() noexcept
+        : m_Override(this)
+    {
+        struct tm tNow;
+        tNow.tm_year = 118; // 2018
+        tNow.tm_mon = 7;    // August
+        tNow.tm_mday = 21;  // 21
+        tNow.tm_hour = 16;  // 4 pm
+        tNow.tm_min = 8;    // 4:08
+        tNow.tm_sec = 43;   // 4:08:43
```

Comment:
I'm going to suggest using aggregate initialization for speed and correctness. This will enforce that all members are initialized in-order. Designations for NRVO (named return value optimization/copy-epsilon) won't exist until C++20 so you'd have to put the object named before the dot. Since you have comments on the end of each line it shouldn't be necessary but might still make things clearer. Using `struct `like that looks kind of strange to me.
Should look like this...
```
std::tm tNow
{
        tNow.tm_sec = 43,   // 4:08:43
        tNow.tm_min = 8;    // 4:08
        tNow.tm_hour = 16;  // 4 pm
        tNow.tm_mday = 21;  // 21
        tNow.tm_mon = 7;    // August
        tNow.tm_year = 118; // 2018
       // The rest are automatically 0
};
```
As you can see, the order of initialization is different than what you had. Without the stuff on the left side, there would be no error, with it there should be (I say should because I haven't really checked if VC++ checks for it, GCC does though). The C++20 version wouldn't have the `tNow `before the dot.